odoo/osv/expression.py

```diff
@@ -1416,6 +1419,12 @@ def push_result(sql):
                         indexed_value = self._unaccent(SQL("jsonb_path_query_array(%s, '$.*')::text", sql_column))
                         _sql_operator = SQL('LIKE') if operator == '=' else sql_operator
                         push_result(SQL("%s %s %s", indexed_value, _sql_operator, self._unaccent(SQL("%s", _right))))
+
+                elif operator in ('in', 'not in') and isinstance(right, str):
+                    dict_op = {'in': 'like', 'not in': 'not like'}
```

Comment:
This is not correct because if you search with "in" it is an equality operator not a like.
In v19, if you do `(x, "in", "abc")` it will convert it to `(x, "in", ["abc"])`.